code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -427,19 +434,36 @@
 	layer = MASSIVE_OBJ_LAYER
 	/// The amount of time the rift has charged for.
 	var/time_charged = 0
-	/// The maximum charge the rift can have.  It actually goes to max_charge + 1, as to prevent constantly retriggering the effects on full charge.
+	/// The maximum charge the rift can have.
 	var/max_charge = 480
 	/// How many carp spawns it has available.
 	var/carp_stored = 0
 	/// A reference to the Space Dragon that created it.
 	var/mob/living/simple_animal/hostile/space_dragon/dragon
+	/// Current charge state of the rift.
+	var/charge_state = CHARGE_ONGOING
+	/// The time since an extra carp was added to the ghost role spawning pool.
+	var/last_carp_inc = 0
 
 /obj/structure/carp_rift/Initialize(mapload)
 	. = ..()
 	carp_stored = 1
-	time_charged = 1
+	icon_state = "carp_rift_carpspawn"
+	set_light_color(LIGHT_COLOR_PURPLE)
+	update_light()
```

Comment:
Why are you setting color in init, instead of on the object's definition?

Reply:
Don't even use initial anywhere in the code for this. I've just set the definition to be the default spawning state.